File: templates/Features/FirstUsePromptVB/Services/FirstRunDisplayService.vb
Imports Param_ItemNamespace.Views
Imports Microsoft.Toolkit.Uwp.Helpers

Namespace Services
    Public NotInheritable Class FirstRunDisplayService
        Shared shown As Boolean = false
        Private Sub New()
        End Sub

        Friend Shared Async Function ShowIfAppropriateAsync() As Task
            If SystemInformation.IsFirstRun And Not shown Then
                shown = true
                Dim dialog = New FirstRunDialog()
                Await dialog.ShowAsync()
            End If
        End Function
    End Class
End Namespace

File: templates/Features/FirstUsePrompt.Prism/Services/FirstRunDisplayService.cs
using Param_ItemNamespace.Views;
using Microsoft.Toolkit.Uwp.Helpers;
using System;
using System.Threading.Tasks;

namespace Param_ItemNamespace.Services
{
    public class FirstRunDisplayService : IFirstRunDisplayService
    {
        private static bool shown = false;

        public async Task ShowIfAppropriateAsync()
        {
            if (SystemInformation.IsFirstRun && !shown)
            {
                shown = true;
                var dialog = new FirstRunDialog();
                await dialog.ShowAsync();
            }
        }
    }
}
